File: cSharp/Api-ng-sample-code/Api-ng-sample-code/TO/MarketBook.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Newtonsoft.Json;

namespace Api_ng_sample_code.TO
{
    public class MarketBook
    {
        [JsonProperty(PropertyName = "marketId")]
        public string MarketId { get; set; }

        [JsonProperty(PropertyName = "isMarketDataDelayed")]
        public bool IsMarketDataDelayed { get; set; }

        [JsonProperty(PropertyName = "status")]
        public MarketStatus Status { get; set; }

        [JsonProperty(PropertyName = "betDelay")]
        public int BetDelay { get; set; }

        [JsonProperty(PropertyName = "bspReconciled")]
        public bool IsBspReconciled { get; set; }

        [JsonProperty(PropertyName = "complete")]
        public bool IsComplete { get; set; }

        [JsonProperty(PropertyName = "inplay")]
        public bool IsInplay { get; set; }

        [JsonProperty(PropertyName = "numberOfWinners")]
        public int NumberOfWinners { get; set; }

        [JsonProperty(PropertyName = "numberOfRunners")]
        public int NumberOfRunners { get; set; }

        [JsonProperty(PropertyName = "numberOfActiveRunners")]
        public int NumberOfActiveRunners { get; set; }

        [JsonProperty(PropertyName = "lastMatchTime")]
        public DateTime? LastMatchTime { get; set; }

        [JsonProperty(PropertyName = "totalMatched")]
        public double TotalMatched { get; set; }

        [JsonProperty(PropertyName = "totalAvailable")]
        public double TotalAvailable { get; set; }

        [JsonProperty(PropertyName = "crossMatching")]
        public bool IsCrossMatching { get; set; }

        [JsonProperty(PropertyName = "runnersVoidable")]
        public bool IsRunnersVoidable { get; set; }

        [JsonProperty(PropertyName = "version")]
        public long Version { get; set; }

        [JsonProperty(PropertyName = "runners")]
        public List<Runner> Runners { get; set; }

        public override string ToString()
        {
            var sb = new StringBuilder().AppendFormat("{0}", "MarketBook")
                        .AppendFormat(" : MarketId={0}", MarketId)
                        .AppendFormat(" : Status={0}", Status)
                        .AppendFormat(" : BetDelay={0}", BetDelay)

                        .AppendFormat(" : IsMarketDataDelayed={0}", IsMarketDataDelayed)
                        .AppendFormat(" : IsBspReconciled={0}", IsBspReconciled)
                        .AppendFormat(" : IsComplete={0}", IsComplete)
                        .AppendFormat(" : IsInplay={0}", IsInplay)
                        .AppendFormat(" : IsCrossMatching={0}", IsCrossMatching)
                        .AppendFormat(" : IsRunnersVoidable={0}", IsRunnersVoidable)

                        .AppendFormat(" : NumberOfWinners={0}", NumberOfWinners)
                        .AppendFormat(" : NumberOfRunners={0}", NumberOfRunners)
                        .AppendFormat(" : NumberOfActiveRunners={0}", NumberOfActiveRunners)

                        .AppendFormat(" : LastMatchTime={0}", LastMatchTime)
                        .AppendFormat(" : TotalMatched={0}", TotalMatched)
                        .AppendFormat(" : TotalAvailable={0}", TotalAvailable)

                        .AppendFormat(" : Version={0}", Version);

            if (Runners != null && Runners.Count > 0)
            {
                int idx = 0;
                foreach (var runner in Runners)
                {
                    sb.AppendFormat(" : Runner[{0}]={1}", idx++, runner);
                }
            }

            return sb.ToString();
        }
    }
}

File: android/src/com/betfair/api/ng/objects/MarketBook.java
package com.betfair.api.ng.objects;

import java.util.Date;
import java.util.List;

import org.apache.commons.lang3.builder.EqualsBuilder;
import org.apache.commons.lang3.builder.HashCodeBuilder;

public class MarketBook {
	    private String marketId;
	    private Boolean isMarketDataDelayed;	
	    private String status;
	    private int betDelay;
	    private Boolean bspReconciled;
	    private Boolean complete;
	    private Boolean inplay;
	    private int numberOfWinners;
	    private int numberOfRunners;
	    private int numberOfActiveRunners;
	    private Date lastMatchTime;
	    private Double totalMatched;
	    private Double totalAvailable;
	    private Boolean crossMatching;
	    private Boolean runnersVoidable;
	    private Long version;
	    private List<Runner> runners;
	    
	    public List<Runner> getRunners() {
			return runners;
		}

		public void setRunners(List<Runner> runners) {
			this.runners = runners;
		}

		public String getMarketId() {
			return marketId;
		}

		public void setMarketId(String marketId) {
			this.marketId = marketId;
		}

		public Boolean getIsMarketDataDelayed() {
			return isMarketDataDelayed;
		}

		public void setIsMarketDataDelayed(Boolean isMarketDataDelayed) {
			this.isMarketDataDelayed = isMarketDataDelayed;
		}

		public String getStatus() {
			return status;
		}

		public void setStatus(String status) {
			this.status = status;
		}

		public int getBetDelay() {
			return betDelay;
		}

		public void setBetDelay(int betDelay) {
			this.betDelay = betDelay;
		}

		public Boolean getBspReconciled() {
			return bspReconciled;
		}

		public void setBspReconciled(Boolean bspReconciled) {
			this.bspReconciled = bspReconciled;
		}

		public Boolean getComplete() {
			return complete;
		}

		public void setComplete(Boolean complete) {
			this.complete = complete;
		}

		public Boolean getInplay() {
			return inplay;
		}

		public void setInplay(Boolean inplay) {
			this.inplay = inplay;
		}

		public int getNumberOfWinners() {
			return numberOfWinners;
		}

		public void setNumberOfWinners(int numberOfWinners) {
			this.numberOfWinners = numberOfWinners;
		}

		public int getNumberOfRunners() {
			return numberOfRunners;
		}

		public void setNumberOfRunners(int numberOfRunners) {
			this.numberOfRunners = numberOfRunners;
		}

		public int getNumberOfActiveRunners() {
			return numberOfActiveRunners;
		}

		public void setNumberOfActiveRunners(int numberOfActiveRunners) {
			this.numberOfActiveRunners = numberOfActiveRunners;
		}

		public Date getLastMatchTime() {
			return lastMatchTime;
		}

		public void setLastMatchTime(Date lastMatchTime) {
			this.lastMatchTime = lastMatchTime;
		}

		public Double getTotalMatched() {
			return totalMatched;
		}

		public void setTotalMatched(Double totalMatched) {
			this.totalMatched = totalMatched;
		}

		public Double getTotalAvailable() {
			return totalAvailable;
		}

		public void setTotalAvailable(Double totalAvailable) {
			this.totalAvailable = totalAvailable;
		}

		public Boolean getCrossMatching() {
			return crossMatching;
		}

		public void setCrossMatching(Boolean crossMatching) {
			this.crossMatching = crossMatching;
		}

		public Boolean getRunnersVoidable() {
			return runnersVoidable;
		}

		public void setRunnersVoidable(Boolean runnersVoidable) {
			this.runnersVoidable = runnersVoidable;
		}

		public Long getVersion() {
			return version;
		}

		public void setVersion(Long version) {
			this.version = version;
		}

		public String toString() {
	    	return "{"+""+"marketId="+getMarketId()+","+"isMarketDataDelayed="+getIsMarketDataDelayed()+","+"status="+getStatus()+","+"betDelay="+getBetDelay()+","+"bspReconciled="+getBspReconciled()+","+"complete="+getComplete()+","+"inplay="+getInplay()+","+"numberOfWinners="+getNumberOfWinners()+","+"numberOfRunners="+getNumberOfRunners()+","+"numberOfActiveRunners="+getNumberOfActiveRunners()+","+"lastMatchTime="+getLastMatchTime()+","+"totalMatched="+getTotalMatched()+","+"totalAvailable="+getTotalAvailable()+","+"crossMatching="+getCrossMatching()+","+"runnersVoidable="+getRunnersVoidable()+","+"version="+getVersion()+","+"runners="+getRunners()+","+"}";
	    }

	    public boolean equals(Object o) {
	        if (!(o instanceof MarketBook)) {
	            return false;
	        }

	        if (this == o) {
	            return true;
	        }
	        MarketBook another = (MarketBook)o;

	        return new EqualsBuilder()
	            .append(marketId, another.marketId)
	            .append(isMarketDataDelayed, another.isMarketDataDelayed)
	            .append(status, another.status)
	            .append(betDelay, another.betDelay)
	            .append(bspReconciled, another.bspReconciled)
	            .append(complete, another.complete)
	            .append(inplay, another.inplay)
	            .append(numberOfWinners, another.numberOfWinners)
	            .append(numberOfRunners, another.numberOfRunners)
	            .append(numberOfActiveRunners, another.numberOfActiveRunners)
	            .append(lastMatchTime, another.lastMatchTime)
	            .append(totalMatched, another.totalMatched)
	            .append(totalAvailable, another.totalAvailable)
	            .append(crossMatching, another.crossMatching)
	            .append(runnersVoidable, another.runnersVoidable)
	            .append(version, another.version)
	            .append(runners, another.runners)
	            .isEquals();
	    }

	    public int hashCode() {
	        return new HashCodeBuilder()
	            .append(marketId)
	            .append(isMarketDataDelayed)
	            .append(status)
	            .append(betDelay)
	            .append(bspReconciled)
	            .append(complete)
	            .append(inplay)
	            .append(numberOfWinners)
	            .append(numberOfRunners)
	            .append(numberOfActiveRunners)
	            .append(lastMatchTime)
	            .append(totalMatched)
	            .append(totalAvailable)
	            .append(crossMatching)
	            .append(runnersVoidable)
	            .append(version)
	            .append(runners)
	            .toHashCode();
	    }

}
